src/ir/module-utils.cpp

```diff
@@ -48,6 +48,7 @@ Function* copyFunction(Function* func,
                        std::optional<std::vector<Index>> fileIndexMap) {
   auto ret = std::make_unique<Function>();
   ret->name = newName.is() ? newName : func->name;
+  ret->hasExplicitName = func->hasExplicitName && !newName.is();
```

Comment:
Why not set it to true when `newName.is()`?

Reply:
`newName` is a generated name. For instance, some names like `outlined-A$func-name` are generated during inlining. But maybe we can consider it is explicit enough?

Reply:
Ah, I see what you mean, good point. I guess it depends on the user expectation in this code. I think that we do want to keep such names, as we are copying a function, so we should copy the property of having an explicit name.